XenModel/Actions/Pool_Patch/GetDiskSpaceRequirementsAction.cs

```diff
@@ -124,22 +134,24 @@ protected override void Run()
             }
 
             // get reclaimable disk space (excluding current patch)
-            long reclaimableDiskSpace = 0;
-            try
-            {
-                var args = new Dictionary<string, string>();
-                if (currentPatch != null)
-                    args.Add("exclude", currentPatch.uuid);
-                 result = Host.call_plugin(Session, Host.opaque_ref, "disk-space", "get_reclaimable_disk_space", args);
-                 reclaimableDiskSpace = Convert.ToInt64(result);
-            }
-            catch (Failure failure)
-            {
-                log.WarnFormat("Plugin call disk-space.get_reclaimable_disk_space on {0} failed with {1}", Host.Name, failure.Message);
+            long reclaimableDiskSpace = 0;
+
+            if (operation != Actions.DiskSpaceRequirements.OperationTypes.autoupdate || availableDiskSpace < requiredDiskSpace)

```

Comment:
Is this "if" trying to avoid a plugin call when availableDiskSpace < requiredDiskSpace? If so, then maybe we should do it regardless of the operation.